ndbench-core/src/main/java/com/netflix/ndbench/core/generators/DefaultDataGenerator.java

```diff
@@ -16,109 +16,115 @@
+    public String getRandomValue()
+    {
         int randomValueIndex = vRandom.nextInt(config.getNumValues());
         return values.get(randomValueIndex);
     }
 
     @Override
-    public Integer getRandomInteger() {
+    public Integer getRandomInteger()
+    {
         return vRandom.nextInt();
     }
+
     @Override
-    public Integer getRandomIntegerValue() {
+    public Integer getRandomIntegerValue()
+    {
         return vRandom.nextInt(config.getNumValues());
     }
 
-
     @Override
     public String getRandomString()
     {
-        return RandomStringUtils.randomAlphanumeric(config.getDataSize());
+        return generateRandomString(getValueSize());
     }
 
-
-    private void initialize(int lowerBound, int upperBound) {
-        if (config.isUseVariableDataSize()) {
-            for (int i = 0; i < config.getNumKeys(); i++) {
-                if(i%1000==0) {
-                    logger.info("Still initializing sample data for variable data size values. So far: "+ i+" /"+config.getNumKeys());
-                }
-                int valueSize = upperBound;
-                if (upperBound > lowerBound) {
-                    valueSize = vvRandom.nextInt(upperBound - lowerBound) + lowerBound;
-                }
-
-                String value = RandomStringUtils.randomAlphanumeric(valueSize);
-
-                values.add(value);
+    private void initialize()
+    {
+        Instant start = Instant.now();
+        for (int i = 0; i < config.getNumValues(); i++)
+        {
+            if (i % 1000 == 0)
+            {
+                logger.info("Still initializing sample data for values. So far: " + i + " /" + config.getNumValues());
             }
+            values.add(generateRandomString(getValueSize()));
         }
+        Instant end = Instant.now();
+        logger.info("Duration to initialize the dataset of random data (ISO-8601 format): " + Duration.between(start, end));
     }
 
-    private void initialize() {
-        for (int i = 0; i < config.getNumValues(); i++) {
-            if(i%1000==0) {
-                logger.info("Still initializing sample data for values. So far: "+ i+" /"+config.getNumValues());
-            }
-            values.add(constructRandomValue());
+    private int getValueSize()
+    {
+        if (config.isUseVariableDataSize())
+        {
+            return vvRandom.nextInt(
+            Math.abs(config.getDataSizeUpperBound() - config.getDataSizeLowerBound()))
+                   + config.getDataSizeLowerBound();
         }
-
+        return config.getDataSize();
     }
 
-    private String constructRandomValue() {
-
-        if (config.isUseStaticData()) {
-            return StaticValue;
-        }
-
-        return getRandomString();
-
+    private void upsertRandomString()
+    {
+        values.add(vRandom.nextInt(config.getNumValues()), generateRandomString(getValueSize()));
     }
 
+    private String generateRandomString(int length)
+    {
```

Comment:
Remove the commented code if we're not using it :-)

Reply:
Wanted to get your inputs on this. hence kept it, but based on our offline discussion, will go with Joe generation since it is somewhat compressable.